frontend/src/profiles/EditProfile.js

```diff
@@ -98,18 +107,26 @@ export default function EditProfile() {
             profileData.birthday,
             profileData.location,
             profileData.skills,
+            profileData.imageName,
             token)
             .then(() => history.push("/profile/"+ profileData.username+ "/edit")).catch(error => console.log(error));
     }
 
+    function handlePictureChange(event) {
+        const imageFile = event.target.files[0];
+        imageFile ? uploadProfileImage(imageFile, token)
+            .then(data => setProfileData({...profileData, imageName: data}))
+            .catch(error => console.log(error)) : setProfileData({...profileData, imageName: profileData.imageName})
```

Comment:
Why the ternay? ´setProfileData({...profileData, imageName: profileData.imageName})´ sets the `imageName` property to the value it was before. If you just want to do it conditionally you can use `imageFile && upload...` or a simple `if`